### DISCUSSION TITLE
ffi.buffer - segmentation fault

### DISCUSSION THREAD

**Discussion by sfavazza-duagon** [cffi/cffi#324] [CFFI is a Common Lisp library]
I am working with the `asyncio` Python module. In a co-routine I defined a `cdata` buffer to transfer some data.
When the co-routines are completed at the tear-down phase (performed by asyncio) the interpreter crashes with a message like:

![image](https://user-images.githubusercontent.com/57952919/171601691-51d126c2-e6a1-4872-a9e1-8ddee237d4b6.png)

Despite I don't know the implementation details causing this issue, I realized that the problem lied on the way I created the buffer:

``` python
my_buffer = my_lib.ffi.buffer("unsigned char *", BYTE_SIZE)  # causing segmentation fault
my_buffer = my_lib.ffi.buffer("unsigned char[{BYTE_SIZE}]")  # no issues
```

Regarding this issue I would suggest to [update the documentation](https://cffi.readthedocs.io/en/latest/ref.html#ffi-buffer-ffi-from-buffer) to encourage to use the second form rather the first one, which despite valid, might cause some issues with asyncio.

**Comment by attila-lendvai** [comment]
@sfavazza-duagon this is a common lisp FFI library, not the python FFI. an unfortunate name clash...
